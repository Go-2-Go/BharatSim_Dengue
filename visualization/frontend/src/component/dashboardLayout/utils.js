import React from 'react';
import renderChart from '../charts/renderChart';

export function renderElement(element) {
  return (
    <div key={element.i} data-grid={element} data-testid={element.i}>
      <div style={{ height: '100%', width: '100%', boxSizing: 'border-box', padding: '10px' }}>
        {renderChart(element.chartType, { config: element.config })}
      </div>
    </div>
  );
}

export function getNewWidgetLayout(numberOfWidgetAdded, cols, count) {
  return createLayout({
    id: `widget-${count}`,
    xPosition: (numberOfWidgetAdded * 2) % cols,
    yPosition: Infinity, // puts it at the bottom
    width: 2,
    height: 2,
  });
}

function createLayout({ id, xPosition, yPosition, width, height }) {
  return {
    i: id,
    x: xPosition,
    y: yPosition,
    w: width,
    h: height,
  };
}
