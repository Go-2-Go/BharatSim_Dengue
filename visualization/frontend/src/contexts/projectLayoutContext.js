import React from 'react';

const projectLayoutContext = React.createContext({
  projectMetadata: {
    name:'',
  },
  selectedDashboardMetadata: {
    name: '',
  },
});

const ProjectLayoutProvider = projectLayoutContext.Provider;

export { projectLayoutContext, ProjectLayoutProvider };
