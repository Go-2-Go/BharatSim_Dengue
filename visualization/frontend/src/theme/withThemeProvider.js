import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme() {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
