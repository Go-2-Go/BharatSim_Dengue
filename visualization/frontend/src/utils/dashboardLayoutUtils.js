/* eslint-disable react/prop-types */

import React from 'react';
import renderChart from '../component/charts/renderChart';

export function renderElement({ layout, chartType, config }) {
  return (
    <div key={layout.i} data-grid={layout} data-testid={layout.i}>
      <div style={{ height: '100%', width: '100%', boxSizing: 'border-box', padding: '10px' }}>
        {renderChart(chartType, { config })}
      </div>
    </div>
  );
}

export function getNewWidgetLayout(numberOfWidgetAdded, cols, count) {
  return createLayout({
    id: `widget-${count}`,
    xPosition: (numberOfWidgetAdded * 2) % cols,
    yPosition: Infinity, // puts it at the bottom
    width: 2,
    height: 2,
  });
}

function createLayout({ id, xPosition, yPosition, width, height }) {
  return {
    i: id,
    x: xPosition,
    y: yPosition,
    w: width,
    h: height,
  };
}
