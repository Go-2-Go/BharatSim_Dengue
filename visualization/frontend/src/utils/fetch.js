import axios from 'axios';
import { httpMethods } from '../constants/fetch';

let shouldShowError;

function commonErrorHandler(err) {
  shouldShowError(true);
}

function initApiConfig({ setError }) {
  shouldShowError = setError;
}

function fetchData({ url, method = httpMethods.GET, headers, data, query, customErrorHandler }) {
  return axios({ url, method, headers, data, params: query })
    .then((res) => res.data)
    .catch((err) => {
      if (customErrorHandler) {
        customErrorHandler(err);
        return;
      }
      commonErrorHandler(err);
    });
}

function uploadData({ url, headers, data, query, customErrorHandler }) {
  return fetchData({
    url,
    method: httpMethods.POST,
    headers,
    data,
    params: query,
    customErrorHandler,
  });
}

export { fetchData, uploadData, initApiConfig };
