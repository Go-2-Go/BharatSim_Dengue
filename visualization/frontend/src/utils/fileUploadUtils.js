const fileUploadedStatus = {
  LOADING: 'loading',
  SUCCESS: 'success',
  ERROR: 'error',
};

function getStatusAndMessageFor(fileUploadStatus, fileName) {
  return {
    [fileUploadedStatus.ERROR]: {
      status: fileUploadedStatus.ERROR,
      message: `Error occurred while unloading ${fileName}`,
    },
    [fileUploadedStatus.SUCCESS]: {
      status: fileUploadedStatus.SUCCESS,
      message: `${fileName} successfully uploaded`,
    },
    [fileUploadedStatus.LOADING]: {
      status: fileUploadedStatus.LOADING,
      message: `uploading ${fileName}`,
    },
  }[fileUploadStatus];
}

export { fileUploadedStatus, getStatusAndMessageFor };
