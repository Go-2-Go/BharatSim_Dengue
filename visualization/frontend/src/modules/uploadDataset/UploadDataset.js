import Box from '@material-ui/core/Box';
import { Typography } from '@material-ui/core';
import React, { useContext, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';
import Stepper from '@material-ui/core/Stepper';
import ImportDataset from './ImportDataset';
import ConfigureDatatype from './ConfigureDatatype';
import ProjectHeader from '../../uiComponent/ProjectHeader';
import { projectLayoutContext } from '../../contexts/projectLayoutContext';
import Modal from '../../uiComponent/Modal';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles((theme) => {
  return {
    configureProjectDataBar: {
      width: '100%',
      display: 'flex',
      justifyContent: 'space-between',
      backgroundColor: theme.colors.grayScale['100'],
      alignItems: 'center',
      padding: theme.spacing(3, 8),
    },
    contentWrapper: {
      backgroundColor: '#FFFFFF',
      border: '1px solid',
      borderColor: `${theme.colors.primaryColorScale['500']}3d`,
      borderRadius: theme.spacing(1),
      margin: theme.spacing(8),
      padding: theme.spacing(8, 10),
    },
  };
});

function getStepContent(
  stepIndex,
  setFile,
  setSchema,
  file,
  schema,
  handleNext,
  setPreviewData,
  setErrorStep,
  previewData,
) {
  return {
    0: (
      <ImportDataset
        setFile={setFile}
        handleNext={handleNext}
        setPreviewData={setPreviewData}
        setErrorStep={setErrorStep}
      />
    ),
    1: (
      <ConfigureDatatype
        onSchemaSelect={setSchema}
        selectedFile={file}
        handleNext={handleNext}
        previewData={previewData}
      />
    ),
  }[stepIndex];
}

function UploadDataset() {
  const classes = useStyles();
  const { projectMetadata } = useContext(projectLayoutContext);
  const [activeStep, setActiveStep] = useState(0);
  const [errorStep, setErrorStep] = useState(undefined);
  const steps = ['Import Data', 'Configure Datatype', 'Upload to Dashboard'];
  const [file, setFile] = useState();
  const [schema, setSchema] = useState();
  const [, setPreviewData] = useState();

  function handleNext() {
    setActiveStep((prevActiveStep) => prevActiveStep + 1);
  }

  return (
    <Box>
      <ProjectHeader>{projectMetadata.name}</ProjectHeader>
      <Box className={classes.configureProjectDataBar}>
        <Typography variant="h6"> Configure Dashboard Data :: Upload Dataset</Typography>
      </Box>
      <Box className={classes.contentWrapper}>
        <Box px={8} pb={8}>
          <Stepper activeStep={activeStep}>
            {steps.map((label, index) => (
              <Step key={label}>
                <StepLabel error={errorStep === index}>{label}</StepLabel>
              </Step>
            ))}
          </Stepper>
        </Box>
        <Box>
          {getStepContent(
            activeStep,
            setFile,
            setSchema,
            file,
            schema,
            handleNext,
            setPreviewData,
            setErrorStep,
          )}
        </Box>
      </Box>
    </Box>
  );
}

export default UploadDataset;
