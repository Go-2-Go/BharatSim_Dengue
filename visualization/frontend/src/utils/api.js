import { contentTypes, httpMethods } from '../constants/fetch';
import { url as serviceURL } from './url';
import { fetchData, uploadData } from './fetch';

function headerBuilder({ contentType }) {
  return {
    'content-type': contentType,
  };
}

function formDataBuilder(data) {
  const formData = new FormData();
  data.forEach(({ name, value }) => formData.append(name, value));
  return formData;
}

const api = {
  saveDashboard: async ({ widgets, layout, dashboardId, name, count }) => {
    return uploadData({
      url: serviceURL.DASHBOARD_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      data: JSON.stringify({
        dashboardData: { widgets, layout, dashboardId, name, count },
      }),
    });
  },

  getAllDashBoard: async () => fetchData({ url: serviceURL.DASHBOARD_URL }),

  uploadFileAndSchema: async ({ file, schema }) =>
    uploadData({
      url: serviceURL.DATA_SOURCES,
      data: formDataBuilder([
        { name: 'datafile', value: file },
        { name: 'schema', value: JSON.stringify(schema) },
      ]),
      headers: headerBuilder({ contentType: contentTypes.FILE }),
    }),

  getCsvHeaders: async (dataSourceId) => fetchData({ url: serviceURL.getHeaderUrl(dataSourceId) }),

  getDatasources: async () => fetchData({ url: serviceURL.DATA_SOURCES }),

  getData: async (datasource, query) =>
    fetchData({ url: serviceURL.getDataUrl(datasource), query }),

  saveProject: async ({ id, ...data }) => {
    const requestObject = {
      url: serviceURL.PROJECT_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
    };
    if (id) {
      return uploadData({
        ...requestObject,
        data: JSON.stringify({ projectData: { ...data, id } }),
        method: httpMethods.PUT,
      });
    }
    return uploadData({
      ...requestObject,
      data: JSON.stringify({ projectData: { ...data } }),
      method: httpMethods.POST,
    });
  },

  getProjects: async () => {
    return fetchData({
      url: serviceURL.PROJECT_URL,
    });
  },
  getProject: async (id) => {
    return fetchData({
      url: serviceURL.getProjectUrl(id),
    });
  },
};

export { api };
