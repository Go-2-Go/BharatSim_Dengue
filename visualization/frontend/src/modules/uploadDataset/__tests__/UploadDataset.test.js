import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import withThemeProvider from '../../../theme/withThemeProvider';

import * as fileUtils from '../../../utils/fileUploadUtils';
import { ProjectLayoutProvider } from '../../../contexts/projectLayoutContext';
import UploadDataset from '../UploadDataset';

jest.spyOn(fileUtils, 'parseCsv').mockImplementation((csvFile, onComplete) => {
  const data = { data: [{ col1: 'row1', col2: 1 }], errors: [] };
  onComplete(data);
});

const ComponentWithProvider = withThemeProvider(() => (
  <ProjectLayoutProvider
    value={{
      projectMetadata: { name: 'project1', id: '123' },
      selectedDashboardMetadata: { name: 'dashboard1' },
    }}
  >
    <UploadDataset />
  </ProjectLayoutProvider>
));

describe('Upload Dataset', () => {
  it('should match snapshot for upload dataset component', () => {
    const { container } = render(<ComponentWithProvider />);

    expect(container).toMatchSnapshot();
  });

  it('should display import file screen for step 1', () => {
    const { getByText } = render(<ComponentWithProvider />);

    expect(getByText('Drag your file here or')).toBeInTheDocument();
  });

  it('should import file and open configure dataset step ', async () => {
    const { getByText, getByTestId, findByText } = render(<ComponentWithProvider />);
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, { target: { files: [new File([''], 'testFile.csv')] } });

    await findByText('DataFile:');

    expect(getByText('testFile.csv')).toBeInTheDocument();
  });
});
