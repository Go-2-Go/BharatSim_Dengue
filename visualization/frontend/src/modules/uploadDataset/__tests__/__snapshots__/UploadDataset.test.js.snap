// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Dataset should match snapshot for upload dataset component 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      class="MuiBox-root MuiBox-root-5 makeStyles-projectNameBar-4"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        project1
      </h5>
    </div>
    <div
      class="MuiBox-root MuiBox-root-6 makeStyles-configureProjectDataBar-1"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
         Configure Dashboard Data :: Upload Dataset
      </h6>
      <div
        class="MuiBox-root MuiBox-root-8 makeStyles-buttonGroup-7"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Cancel
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Upload
          </span>
        </button>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-9 makeStyles-contentWrapper-2"
    >
      <div
        class="MuiBox-root MuiBox-root-10"
      >
        <div
          class="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiPaper-elevation0"
        >
          <div
            class="MuiStep-root MuiStep-horizontal"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    1
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Import Data
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    2
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Configure Datatype
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    3
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Upload to Dashboard
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-11"
      >
        <div
          class="MuiBox-root MuiBox-root-14 makeStyles-importDataContainer-12"
        >
          <img
            alt="Import File"
            src="importFileIcon.svg"
          />
          <div
            class="MuiBox-root MuiBox-root-15 makeStyles-textContainer-13"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Drag your file here or 
            </p>
            <div
              class="MuiBox-root MuiBox-root-16"
            >
              <label
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  Browse
                  <input
                    data-testid="file-input"
                    style="display: none;"
                    type="file"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
